--- a/esvo_core/src/core/EventBM.cpp
+++ b/esvo_core/src/core/EventBM.cpp
@@ -89,7 +89,7 @@
   if(x_rect(0) < 0 || x_rect(0) > camSysPtr_->cam_left_ptr_->width_ - 1 ||
      x_rect(1) < 0 || x_rect(1) > camSysPtr_->cam_left_ptr_->height_ - 1)
     return false;
   // This is to avoid depth estimation happening in the mask area.
-  if(camSysPtr_->cam_left_ptr_->UndistortRectify_mask_(x_rect(1), x_rect(0)) <= 125)
+  if(camSysPtr_->cam_left_ptr_->UndistortRectify_mask_(static_cast<int>(x_rect(1)), static_cast<int>(x_rect(0))) <= 125)
     return false;
   Eigen::Vector2i x1(std::floor(x_rect(0)), std::floor(x_rect(1)));
   Eigen::Vector2i x1_left_top;
--- a/esvo_core/src/core/RegProblemLM.cpp
+++ b/esvo_core/src/core/RegProblemLM.cpp
@@ -383,13 +383,13 @@
       patchCentreCoord(1) < (wy-1)/2 ||
       patchCentreCoord(1) > camSysPtr_->cam_left_ptr_->height_ - (wy-1)/2 - 1)
     return false;
-  if(mask(patchCentreCoord(1)-(wy-1)/2, patchCentreCoord(0)-(wx-1)/2) < 125)
+  if(mask(static_cast<int>(patchCentreCoord(1)-(wy-1)/2), static_cast<int>(patchCentreCoord(0)-(wx-1)/2)) < 125)
     return false;
-  if(mask(patchCentreCoord(1)-(wy-1)/2, patchCentreCoord(0)+(wx-1)/2) < 125)
+  if(mask(static_cast<int>(patchCentreCoord(1)-(wy-1)/2), static_cast<int>(patchCentreCoord(0)+(wx-1)/2)) < 125)
     return false;
-  if(mask(patchCentreCoord(1)+(wy-1)/2, patchCentreCoord(0)-(wx-1)/2) < 125)
+  if(mask(static_cast<int>(patchCentreCoord(1)+(wy-1)/2), static_cast<int>(patchCentreCoord(0)-(wx-1)/2)) < 125)
     return false;
-  if(mask(patchCentreCoord(1)+(wy-1)/2, patchCentreCoord(0)+(wx-1)/2) < 125)
+  if(mask(static_cast<int>(patchCentreCoord(1)+(wy-1)/2), static_cast<int>(patchCentreCoord(0)+(wx-1)/2)) < 125)
     return false;
   return true;
 }
